### PR TITLE
🧪 Scout: fix Objective-C %@ placeholder detection

### PR DESCRIPTION
**Pull request by cungminh2710**
💡 What: Fixed detection of Objective-C style `%@` placeholders in `extra_placeholders.go` and added a new unit test suite `extra_placeholders_scout_test.go`.

🎯 Why: The previous regex used a word boundary `\b` after `@`, which is not a word character in Go. This caused `%@` to be missed when followed by anything other than a word character (like `!`, `.`, or spaces), leading to false-positive validation failures.

🧩 Scope: `internal/i18n/segmentvalidate/extra_placeholders.go` and a new test file `internal/i18n/segmentvalidate/extra_placeholders_scout_test.go`.

✅ Verification: Ran `go test -v ./internal/i18n/segmentvalidate/...`. All tests passed, including the new Objective-C specific cases.

🛡️ Confidence: High. The fix specifically addresses the non-word character boundary issue while preserving existing alphanumeric placeholder protection. Verified with multiple edge cases in the new test suite.

---
*PR created automatically by Jules for task [14969994150260115738](https://jules.google.com/task/14969994150260115738) started by @cungminh2710*

### CODE REVIEW

File: internal/i18n/segmentvalidate/extra_placeholders.go [hunk 1]
```diff
@@ -11,7 +11,8 @@ import (
 var extraPlaceholderPatterns = []*regexp.Regexp{
 	regexp.MustCompile(`%[0-9]+\$[sdf@]`),
 	regexp.MustCompile(`%\([A-Za-z_][\w]*\)[sdf@]`),
-	regexp.MustCompile(`%[sdf@]\b`),
+	regexp.MustCompile(`%[sdf]\b`),
+	regexp.MustCompile(`%@`),
```
[the remark this answers is not in the document]
Comment (cungminh2710) [reply internal/i18n/segmentvalidate/extra_placeholders.go:15]: @jules fix

**Comment by cungminh2710** [comment]
@jules fix merge main conflict

**Comment by cungminh2710** [comment]
@cursoragent fix merge main conflict

File: internal/i18n/segmentvalidate/extra_placeholders.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cungminh2710) [reply internal/i18n/segmentvalidate/extra_placeholders.go:15]: @cursoragent fix
